libs/template/reader.go

```diff
@@ -72,6 +72,33 @@ func (r *builtinReader) LoadSchemaAndTemplateFS(ctx context.Context) (*jsonschem
 
 func (r *builtinReader) Cleanup(ctx context.Context) {}
 
+// overridingReader wraps another Reader and overrides their properties
+type overridingReader struct {
+	underlying               Reader
+	propertyDefaultOverrides map[string]any
+}
+
+func (r *overridingReader) LoadSchemaAndTemplateFS(ctx context.Context) (*jsonschema.Schema, fs.FS, error) {
+	schema, schemaFS, err := r.underlying.LoadSchemaAndTemplateFS(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for propName, newDefault := range r.propertyDefaultOverrides {
+		if propSchema, ok := schema.Properties[propName]; ok {
+			propSchema.Default = newDefault
+		} else {
+			return nil, nil, fmt.Errorf("cannot override default for non-existing property %s", propName)
+		}
+	}
+
+	return schema, schemaFS, nil
+}
+
+func (r *overridingReader) Cleanup(ctx context.Context) {
+	r.underlying.Cleanup(ctx)
+}
+
```

Comment:
Is this usable for other templates?

@lennartkats-db added something to reuse templates from another directory and pre-populate values from the template inputs definition. Seems related.